awx/ui/client/lib/models/UnifiedJobTemplate.js

```diff
@@ -0,0 +1,110 @@
+let BaseModel;
+let WorkflowJobTemplateNode;
+let $http;
+
+function optionsLaunch (id) {
```

Comment:
I think there should be some way to tie in these kinds of sub-resource actions to the base model. Part of the goal with the models is that if all interaction with the API flows throw the base model, then all parsing/generation of query string parameters, payloads, etc will be universal and consistent, and it opens up the opportunity for easy caching, rbac, session management, error handling, pagination, and all that.

I haven't run into this case (yet) with the model work that I've done, but I think it'd be good kick around some ideas here if you're up for it. Interface-wise, it could be as straightforward as something like:

```
workflowJobTemplateNode.request({
    method: 'options',
    subResource: 'launch'
})
```

But handling sub-resources in the base model (parsing the request config argument, building the URL, and storing data where applicable) would have to be implemented.

Reply:
For what it's worth, I like the idea of having a generalized way to tie into the action endpoints of resources - though we could maybe limit the `subResource` to the actual `related` link fields on the response data to avoid any explicit url building.

As an initial implementation, I'm wondering if we need to worry about caching / storing response data at all for these linked items or if we could make that something the caller needs to handle if they so choose.

Reply:
Another thing I think makes most sense to wait for @mabashian's changes.